ruby/ql/lib/codeql/ruby/ast/Pattern.qll

```diff
@@ -97,3 +97,302 @@ class TuplePattern extends Pattern, TTuplePattern {
 
   override AstNode getAChild(string pred) { pred = "getElement" and result = this.getElement(_) }
 }
+
+private class TPatternNode =
+  TArrayPattern or TFindPattern or THashPattern or TAlternativePattern or TAsPattern or
+      TVariableReferencePattern;
+
+private class TPattern =
+  TPatternNode or TLiteral or TLambda or TConstantAccess or TLocalVariableAccess or
```

Comment:
Why is `TLambda` included?

Reply:
I see that you have a test case `in -> x { x == 10 }`, so I guess that is why. But there are no sub classes of `CasePattern` that correspond to this case; should there be?

Reply:
We could add a sub class. Perhaps we should do that for all expressions that can be used in a pattern (constant, literal, lambda, etc).

Reply:
I think that would be nice. I have no clue what a lambda pattern does.